### ISSUE TITLE
Lint for unused widgets and widget parameters

### ISSUE THREAD

**Issue by rasitayaz** [dart-lang/linter#3552]
It would be great to be able to see if a widget/widget property is unnecessary and not used.

**Examples**
Let's say I've created a widget and decided not to use it at some point in the app's development, but forgot to delete it from the project. So it will stay there forever unless I come across by chance. Same thing with the widget properties.

**Comment by incendial** [comment]
@rasitayaz hi, if you're interested in checking for unused code (like unused declared widgets), you can try https://dartcodemetrics.dev/docs/cli/check-unused-code (we also provide other commands for code refactoring)

**Comment by rasitayaz** [comment]
thank you, couldn't find this as a rule on https://dart-lang.github.io/linter/lints/index.html. can i enable this as a linter rule in my `analysis_options.yaml` to display warnings if there is any unused code?

**Comment by incendial** [comment]
Nope, it's a separate package and this is a cli command. Running such check as linter rule will be quite memory expensive and might make the overall dev experience worse.
